packages/commands/src/solana/deposit.ts

```diff
@@ -0,0 +1,210 @@
+      });
+
+      if (!matchingTokenAccount) {
+        throw new Error(`No token account found for mint ${options.mint}`);
+      }
+
+      // Check token balance
+      const accountInfo = await connection.getTokenAccountBalance(
+        matchingTokenAccount.pubkey
+      );
+      const balance = accountInfo.value.uiAmount;
+      const amountToSend = parseFloat(options.amount);
+      if (!balance || balance < amountToSend) {
+        throw new Error(
+          `Insufficient token balance. Available: ${
+            balance ?? 0
+          }, Required: ${amountToSend}`
+        );
+      }
+
+      const from = matchingTokenAccount.pubkey;
+
+      // Find the TSS PDA (meta)
+      const [tssPda] = PublicKey.findProgramAddressSync(
+        [Buffer.from("meta", "utf-8")],
+        gatewayProgram.programId
+      );
+
+      // Find the TSS's ATA for the mint
+      const tssAta = await PublicKey.findProgramAddress(
+        [
+          tssPda.toBuffer(),
+          TOKEN_PROGRAM_ID.toBuffer(),
+          new PublicKey(options.mint).toBuffer(),
+        ],
+        ASSOCIATED_TOKEN_PROGRAM_ID
+      );
+
+      const to = tssAta[0].toBase58();
+
+      const tx = await gatewayProgram.methods
+        .depositSplToken(
+          new anchor.BN(ethers.parseUnits(options.amount, decimals).toString()),
+          receiverBytes,
+          null
+        )
+        .accounts({
+          from,
+          mintAccount: options.mint,
+          signer: keypair!.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+          to,
+          tokenProgram: options.tokenProgram,
+        })
+        .rpc();
+      console.log("Transaction hash:", tx);
+    } else {
+      // Check SOL balance
+      const balance = await connection.getBalance(keypair!.publicKey);
+      const lamportsNeeded = ethers.parseUnits(options.amount, 9).toString();
+      if (balance < parseInt(lamportsNeeded)) {
+        throw new Error(
+          `Insufficient SOL balance. Available: ${balance / 1e9}, Required: ${
+            options.amount
+          }`
+        );
+      }
+      const tx = await gatewayProgram.methods
+        .deposit(
+          new anchor.BN(ethers.parseUnits(options.amount, 9).toString()),
+          receiverBytes,
+          null
+        )
+        .accounts({})
+        .rpc();
+      console.log("Transaction hash:", tx);
+    }
+  } catch (error) {
+    handleError({
+      context: "Error during deposit",
+      error,
+      shouldThrow: false,
+    });
+    process.exit(1);
+  }
+};
+
+export const depositCommand = new Command("deposit")
+  .description("Deposit tokens from Solana")
+  .requiredOption("--amount <amount>", "Amount of tokens to deposit")
+  .requiredOption("--recipient <recipient>", "Recipient address")
+  .addOption(
+    new Option("--mnemonic <mnemonic>", "Mnemonic").conflicts(["private-key"])
+  )
+  .addOption(
+    new Option(
+      "--private-key <privateKey>",
+      "Private key in base58 format"
```

Comment:
The result of the `accounts show` command returns a pk in hex format, but this command requires a base58 pk, while it surely is a simple conversion it is still inconvenient.